man/tpm2_createek.1.md

```diff
@@ -42,10 +42,15 @@ Refer to:
   * **-G**, **\--key-algorithm**=_ALGORITHM_:
 
     The endorsement key algorithm. Supports:
-    * **ecc** - An P256 key.
+    * **ecc** - A NIST_P256 key by default. Alternative ECC curves can be selected
+      with the **-e** option.
     * **rsa** - An RSA2048 key.
     * **keyedhash** - hmac key.
 
+  * **-e**, **\--ecc-curve**=_ECC_CURVE_ID_:
```

Comment:
there is an [algorithm-parser ](https://github.com/tpm2-software/tpm2-tools/blob/master/lib/tpm2_alg_util.c#L313:#L346)for handling ECC curves so we don't have to add another option.

Reply:
Thanks for the hint. But how do can the non-NIST ECC curve **SM2_P256** be parsed for which an EK template exists? Can the algorithm-parser be extended?

Reply:
@strongX509 yes you can extend the algorithm-parser as long as the curve is defined in the tss.  Looks like 25519 isn't yet added to the tss. @AndreasFuchsSIT do you know?